src/frontend/components/Sidebar/Sidebar.js

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { useLocation } from "react-router-dom";
+import { ExpandedSidebar, ShrinkedSidebar } from "..";
+import { useSidebar } from "../../contexts";
+import { useWindowSize } from "../../hooks";
+import "./Sidebar.css";
+
+export const Sidebar = () => {
+  const { pathname } = useLocation();
+
+  const sidebarForbiddenPaths = ["/", "/signin", "/signup"];
+  const { showSidebar } = useSidebar();
+
+  const windowSize = useWindowSize();
+
+  return (
+    !sidebarForbiddenPaths.includes(pathname) &&
+    (showSidebar ? (
+      <div className="sidebar">
+        <ExpandedSidebar />
+      </div>
+    ) : (
+      windowSize.width > 950 && (
+        <div className="sidebar">
```

Comment:
If this div is common take it outside on line 18 and then use the condition inside to avoid duplicating the markup.